RecoTauTag/HLTProducers/interface/L1TJetsMatching.h

```diff
@@ -111,26 +112,133 @@ std::pair<std::vector<T>, std::vector<T>> categorise(
+        if (i == i1)
+          continue;
+        if (i == i2)
+          continue;
+        if (!filledi3) {
+          i3 = i;
+          filledi3 = true;
+        } else {
+          i4 = i;
+          break;
+        }
+      }
+      const T& myJet3 = (pfMatchedJets)[i3];
+      const T& myJet4 = (pfMatchedJets)[i4];
+      vec4jets.push_back(myJet3);
+      vec4jets.push_back(myJet4);
+    }
+
+    if ((mjj > Mjj) && (myJet1.pt() < pt1) && (myJet1.pt() < pt3) && (myJet1.pt() > pt2) &&
+        (myJet2.pt() > pt2)) {  //60, 30, 50, 500
+
+      std::vector<unsigned int> idx_jets;
+      idx_jets.clear();
+
+      for (unsigned int i = 1; i < pfMatchedJets.size(); i++) {  //skip highest pT jet
+        if (i == i1 || i == i2)
+          continue;
+        if ((pfMatchedJets)[i].pt() > pt2) {
+          idx_jets.push_back(i);
+        }
+      }
+      if (idx_jets.size() == 2) {
+        const T& thirdjet = (pfMatchedJets)[0];
+        const T& fourthjet = (pfMatchedJets)[idx_jets.at(0)];
+        const T& fifthjet = (pfMatchedJets)[idx_jets.at(1)];
+
+        vec5jets.push_back(myJet1);
+        vec5jets.push_back(myJet2);
+        vec5jets.push_back(thirdjet);
+        vec5jets.push_back(fourthjet);
+        vec5jets.push_back(fifthjet);
+      } else if (idx_jets.size() > 2) {
+        const T& thirdjet = (pfMatchedJets)[0];
+        const T& fourthjet = (pfMatchedJets)[idx_jets.at(0)];
+        const T& fifthjet = (pfMatchedJets)[idx_jets.at(1)];
+        const T& sixthjet = (pfMatchedJets)[idx_jets.at(2)];
+
+        vec6jets.push_back(myJet1);
+        vec6jets.push_back(myJet2);
+        vec6jets.push_back(thirdjet);
+        vec6jets.push_back(fourthjet);
+        vec6jets.push_back(fifthjet);
+        vec6jets.push_back(sixthjet);
+      }
+    }
+
+    output = std::make_tuple(vec4jets, vec5jets, vec6jets);
+  }
+
+  return output;
+}
+
 template <typename T>
 L1TJetsMatching<T>::L1TJetsMatching(const edm::ParameterSet& iConfig)
     : jetSrc_(consumes<std::vector<T>>(iConfig.getParameter<edm::InputTag>("JetSrc"))),
       jetTrigger_(consumes<trigger::TriggerFilterObjectWithRefs>(iConfig.getParameter<edm::InputTag>("L1JetTrigger"))),
+      matchingMode_(iConfig.getParameter<std::string>("matchingMode")),
       pt1Min_(iConfig.getParameter<double>("pt1Min")),
       pt2Min_(iConfig.getParameter<double>("pt2Min")),
       pt3Min_(iConfig.getParameter<double>("pt3Min")),
       mjjMin_(iConfig.getParameter<double>("mjjMin")),
       matchingR_(iConfig.getParameter<double>("matchingR")),
       matchingR2_(matchingR_ * matchingR_) {
-  produces<std::vector<T>>("TwoJets");
-  produces<std::vector<T>>("ThreeJets");
+  if (matchingMode_ == "VBF") {  // Default
+    produces<std::vector<T>>("TwoJets");
+    produces<std::vector<T>>("ThreeJets");
+  } else if (matchingMode_ == "VBFPlus2CentralJets") {
+    produces<std::vector<T>>("FourJets");
+    produces<std::vector<T>>("FiveJets");
+    produces<std::vector<T>>("SixJets");
+  }
```

Comment:
Sorry, my question was not clear: I understand the use of throwing an exception, but I don't understand what you meant by "please don't integrate this suggestion via GitHub"?

Reply:
When suggestions are provided in GitHub comments, they can be committed from the webpage, see [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request).

Reply:
I see, thanks! Just to be completely clear: doing the change via the webpage is what I should *NOT* do, right?

Reply:
Correct, because in this case I wrote the suggestion ignoring code-checks, so code-checks would have almost certainly failed applying the suggestion literally.

Reply:
Thanks, now throwing an exception passing code-format.